### DISCUSSION TITLE
docker images >= v0.3.11 do not work on armv8/Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by djthorpe** [open-webui/open-webui#4530] [General]
# Bug Report

## Installation Method

On a Raspberry Pi 4 the following fails:

```bash
> docker run ghcr.io/open-webui/open-webui:main

Loading WEBUI_SECRET_KEY from file, not provided as an environment variable.
Generating WEBUI_SECRET_KEY
Loading WEBUI_SECRET_KEY from .webui_secret_key
USER_AGENT environment variable not set, consider setting it to identify your requests.

[docker then fails with exit code 132]
```

The docker exit code is 132, which I believe means invalid opcode exception, indicating that the docker container was compiled with ARM instructions which are not supported by the Raspberry Pi 4 instruction set (ISA).

However, the v0.3.10 image runs:

```bash
docker run ghcr.io/open-webui/open-webui:git-c74e7df
```

This indicates some change in downstream requirements. Between v0.3.10 and v0.3.11 the following `backend/requirements.txt` were changed:

```
google-generativeai==0.5.4 => google-generativeai==0.7.2
langchain==0.2.6 => langchain==0.2.11
langchain-community==0.2.6 => langchain-community==0.2.10
unstructured==0.14.10 => unstructured==0.15.0
rapidocr-onnxruntime==1.3.22 => rapidocr-onnxruntime==1.3.24
langfuse==2.38.0 => langfuse==2.39.2
duckduckgo-search~=6.1.12 => duckduckgo-search~=6.2.1
```

Not sure it's one of these, or some other changed downstream requirement elsewhere however.

## Environment

- **Open WebUI Version:** v0.3.11 and subsequent versions
- **Operating System:** Debian GNU/Linux 12 (bookworm)

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Docker container runs, and server starts.

## Actual Behavior:

Docker container quits before server starts, with 132 exit code (invalid opcode)

## Description

**Bug Summary:**
As above - Raspberry 4 (and likely earlier Raspberry Pi's) are no longer supported by the docker image. Likely Raspberry Pi 5 would work (although I am not sure what instruction is being executed). I have also tested on an NVIDIA Jetson NX Orin and this works fine.

## Reproduction Details

**Steps to Reproduce:**

See above

## Logs and Screenshots

**Browser Console Logs:**
N/A

**Docker Container Logs:**
```
No logs are created, due to container exiting before start.
```

**Comment by Monteth** [comment]
Hey, did you try the newer versions? I'm wondering if it was fixed as silently as it was introduced ;)

**Reply by tobu480** [reply]
Just tried "latest" and can confirm that v0.3.30 is working on my rp4

**Reply by Monteth** [reply]
Thank you. I migrated my setup from v0.3.10 to v0.3.30 and confirmed it works flawlessly 👌🏻
